fix fetching a helm chart via oci

DennisDenuto opened the pull request:
Fixes: https://github.com/vmware-tanzu/carvel-vendir/issues/144

Note:
- this PR is based off https://github.com/vmware-tanzu/carvel-vendir/pull/166

DennisDenuto committed:
bug fix: Allow syncing a helm chart using OCI

Authored-by: Dennis Leon <leonde@vmware.com>

DennisDenuto committed:
test refactor: backfill helm chart with dependencies test

- also update the vendor assets under examples and fix the examples_test
  git diff assertions. (It was running under the wrong directory and was
  giving false positive test results)

Authored-by: Dennis Leon <leonde@vmware.com>

cppforlife commented on test/e2e/example.go:
```diff
@@ -4,16 +4,20 @@
+	Env               []string
+	OnlyLocked        bool
+	SkipRemove        bool
+	YttVendirYamlArgs []string
```
neat! though for the sake of minimizing dependencies, should we just do string replace (e.g. vendir.yml can have REPLACE_ME_REGISTRY_ADDR)?

DennisDenuto replied:
ok fine.... sounds reasonable!

DennisDenuto replied:
ok done

DennisDenuto committed:
test refactor: remove ytt as a test dependency

- instead use golang ReplaceAll function to inject test registry into
  vendir.yaml file

Authored-by: Dennis Leon <leonde@vmware.com>
